### ISSUE TITLE
[BUG] TLS versions are ignored

### ISSUE THREAD

**Issue by ZingyAwesome** [linuxserver/docker-swag#365]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After the latest update of SWAG (2.5.0-ls200), the TLS version configuration is ignored. I believe this is a bug caused by NGINX, where the `ssl_protocols` directive is ignored in any `server {}` blocks.
This causes the TLS version to fall back to 1.2, and TLS 1.3 becomes disabled.

Before the last update, the SSL configuration was `include`d in the main nginx.conf file, and this worked. Now it is not, and is only included in `server {}` blocks, triggering this bug. I have not seen any errors in logs caused by this bug however.

There are similar reports, see [this](https://github.com/NginxProxyManager/nginx-proxy-manager/issues/2822)

### Expected Behavior

The `ssl_protocols` directive should be respected and TLS 1.3 should be enabled.
This could be solved by moving said directive back to nginx.conf, at least until the bug is fixed. This would leave part of the SSL configuration in ssl.conf and part in nginx.conf which isn't desirable, unless you can think of another solution.

### Steps To Reproduce

1. Update to the latest version of SWAG
2. Make sure the nginx.conf and site-confs/default.conf have been updated to the latest version
3. Use a SSL/TLS testing website, it will report that TLS 1.3 has been disabled even though `ssl_protocols TLSv1.2 TLSv1.3;` is present by default in ssl.conf

### Environment

```markdown
- OS: Ubuntu 22.04 LTS
- How docker service was installed: APT
```

### CPU architecture

x86-64

### Docker creation

```bash
services:
  swag:
    image: lscr.io/linuxserver/swag
    container_name: swag
    cap_add:
      - NET_ADMIN
    env_file:
      - stack.env
    networks:
      - swag
    ports:
      - "443:443"
      - "80:80"
    volumes:
      - /opt/docker/swag/config:/config
    logging:
      driver: local
    restart: always


networks:
  swag:
    external: true
```

### Container logs

```bash
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗ 
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝ 

   Brought to you by linuxserver.io
───────────────────────────────────────

To support the app dev(s) visit:
Certbot: https://supporters.eff.org/donate/support-work-on-certbot

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    950
User GID:    950
───────────────────────────────────────

using keys found in /config/keys
Variables set:
PUID=950
PGID=950
TZ=UTC
URL=<my-domain>
SUBDOMAINS=wildcard
EXTRA_DOMAINS=
ONLY_SUBDOMAINS=false
VALIDATION=dns
CERTPROVIDER=letsencrypt
DNSPLUGIN=cloudflare
EMAIL=<my-email>
STAGING=false

Using Let's Encrypt as the cert provider
SUBDOMAINS entered, processing
Wildcard cert for <my-domain> will be requested
E-mail address entered: <my-email>
dns validation via cloudflare plugin is selected
Certificate exists; parameters unchanged; starting nginx
The cert does not expire within the next day. Letting the cron script handle the renewal attempts overnight.
**** Configuring CrowdSec nginx Bouncer ****
**** Successfully configured CrowdSec nginx Bouncer v1.0.5 ****
[mod-init] **** Installing all mod packages ****
fetch http://dl-cdn.alpinelinux.org/alpine/v3.17/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.17/community/x86_64/APKINDEX.tar.gz
OK: 212 MiB in 219 packages
[custom-init] No custom files found, skipping...
[ls.io-init] done.
Server ready
nginx: [alert] [lua] init_by_lua:8: [Crowdsec] Initialisation done
```

**Comment by nemchik** [comment]
As noted in https://github.com/NginxProxyManager/nginx-proxy-manager/issues/2822

https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_protocols documents that `ssl_protocols` should be valid in `server` context.

I'm guessing there is an actual legitimate bug in nginx itself. I can't find any currently open bug reports or issues pointing to this being a problem in nginx (although I didn't dig too deep at https://trac.nginx.org/nginx/ ).

**Comment by nemchik** [comment]
After testing my own instances with https://www.ssllabs.com/ssltest/ I am unable to reproduce the issue of TLS 1.3 not working with the default configs we ship.

I tested:

- Intentionally disabling TLSv1.3 and then re-enabling it: worked as expected
- Including `ssl.conf` exclusively in `nginx.conf`: worked as expected
- Including `ssl.conf` exclusively in `site-confs/default.conf`: worked as expected
- Including `ssl.conf` in both `nginx.conf` and `site-confs/default.conf`: worked as expected

As a side note: all my other configs are update (no warnings in the log about outdated configs).

I also confirmed with @drizuid in our discord that he gets the same result (everything works as expected with the default shipped configs fully updated).

Because we cannot reproduce the issue, and cannot find any open issues/bug reports with nginx confirming this problem, I'm going to close this issue.

The only thing I can think that could be causing the problem for you is if you are using Cloudflare's DNS proxy and you do not have TLS 1.3 enabled there.
